File: MoviesAPIClient/Features/TMDB/Models/SearchResult.swift
//
//  SearchResult.swift
//  MoviesAPIClient
//
//  Created by Aryan Rogye on 9/14/26.
//

enum MediaType: String, Decodable {
    case tv
    case movie
    case person
}

struct SearchResult: Decodable, Identifiable {
    let id: Int
    let mediaType: MediaType
    let title: String?
    let name: String?
    let posterPath: String?
    let overview: String?
}

struct SearchResponse: Decodable {
    let results: [SearchResult]
}

File: MoviesShared/sharedLogic/src/commonMain/kotlin/com/aryanrogye/movies_shared/models/SearchResult.kt
package com.aryanrogye.movies_shared.models

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
enum class KTMediaType(val rawValue: String) {
    @SerialName("tv")
    TV("tv"),

    @SerialName("movie")
    MOVIE("movie"),

    @SerialName("person")
    PERSON("person")
}

@Serializable
data class KTSearchResult(
    val id: Int,
    val mediaType: KTMediaType,
    val title: String? = null,
    val name: String? = null,
    val posterPath: String? = null,
    val overview: String? = null,
    val releaseDate: String? = null,
    val firstAirDate: String? = null,
)

@Serializable
data class KTSearchResponse(
    val results: List<KTSearchResult>
)